cache.py

```diff
@@ -87,73 +89,73 @@ def keyFromArgs(f, userSensitive, languageSensitive, evaluatedArgs, path, args,
 	try:
 		appVersion = currentRequest.get().request.environ["CURRENT_VERSION_ID"].split('.')[0]
 	except:
-		appVersion = ""
 		# FIXME
-		#logging.error("Could not determine the current application id! Caching might produce unexpected results!")
+		logging.error("Could not determine the current application version! Caching might produce unexpected results!")
+		appVersion = ""
 	res["__appVersion"] = appVersion
 	# Last check, that every parameter is satisfied:
 	if not all([x in res.keys() for x in argsOrder]):
-		# we have too few paramerts for this function; that wont work
+		# we have too few parameters for this function; that wont work
 		return None
-	res = list(res.items())  # Flatn our dict to a list
+	res = list(res.items())  # flatten our dict to a list
 	res.sort(key=lambda x: x[0])  # sort by keys
 	mysha512 = sha512()
 	mysha512.update(str(res).encode("UTF8"))
-	return (mysha512.hexdigest())
+	return mysha512.hexdigest()
 
 
-def wrapCallable(f, urls, userSensitive, languageSensitive, evaluatedArgs, maxCacheTime):
+def wrapCallable(f, urls: List[str], userSensitive: int, languageSensitive: bool,
+				 evaluatedArgs: List[str], maxCacheTime: int):
 	"""
 		Does the actual work of wrapping a callable.
 		Use the decorator enableCache instead of calling this directly.
 	"""
 
 	@wraps(f)
-	def wrapF(self, *args, **kwargs):
+	def wrapF(self, *args, **kwargs) -> str:
```

Comment:
This should be Union[str,byte] - it's possible to cache images/pdfs (that's why the content-type is also restored)

Reply:
I think you mean `bytes` - I've corrected it :)